AACT-603: Create design_groups_data method in v2 processor

Javier-Jimenez-18 opened the pull request:
- Created design_groups_data mapper method in the v2 processor.
- Added new column group_intervention_names to design_groups table and updated design_groups_data method.
- Created RSpec test case to test the design_groups_data method.
- All 7 Test Cases PASS in processor_v2_spec.rb:

<img width="1239" alt="Screen Shot 2023-12-13 at 4 14 20 PM" src="https://github.com/ctti-clinicaltrials/aact/assets/81119399/dc304a0c-b74a-459f-9c79-c489780b7809">

micronix reviewed (changes requested):
Undo the addition of the group_intervention_names.

1. Rollback the migration
2. Delete the migration
3. Remove the column from the tests

In addition remove the config/connections.yml file

Javier-Jimenez-18 commented:
- Undid the addition of the group_intervention_names column
- Rollback the migration and deleted the migration file
- Removed the column from method design_groups_data and from the RSpec test
- Removed the config/connections.yml file

Test case for method design_groups_data PASS and processor_v2_spec file PASS:

<img width="877" alt="Screen Shot 2023-12-15 at 11 03 01 PM" src="https://github.com/ctti-clinicaltrials/aact/assets/81119399/48990a74-6fcd-4a4a-83a0-5d10d26db2c6">

micronix reviewed (approved):
Looks good
